src/main/java/mate/academy/spring/dao/impl/RoleDaoImpl.java

```diff
@@ -0,0 +1,30 @@
+package mate.academy.spring.dao.impl;
+
+import java.util.Optional;
+import mate.academy.spring.dao.AbstractDao;
+import mate.academy.spring.dao.RoleDao;
+import mate.academy.spring.exception.DataProcessingException;
+import mate.academy.spring.model.Role;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.query.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class RoleDaoImpl extends AbstractDao<Role> implements RoleDao {
+    public RoleDaoImpl(SessionFactory factory) {
+        super(factory, Role.class);
+    }
+
+    @Override
+    public Optional<Role> getRoleByName(String roleType) {
+        try (Session session = factory.openSession()) {
+            Query<Role> query = session.createQuery(
+                    "FROM Role WHERE roleType = :role", Role.class);
+            query.setParameter("role", Role.RoleType.values());
```

Comment:
`Role.RoleType.values()` return RoleType[]
And you should  set here `String roleType` from method parameters to request